tests/zfs-tests/tests/functional/projectquota/setup.ksh

```diff
@@ -0,0 +1,65 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
+# Use is subject to license terms.
+#
+
+#
+# Copyright (c) 2017 by Fan Yong. All rights reserved.
+#
+
+. $STF_SUITE/tests/functional/projectquota/projectquota_common.kshlib
+
+verify_runnable "both"
+
+del_user $PUSER
+del_group $PGROUP
+log_must add_group $PGROUP
+log_must add_user $PGROUP $PUSER
+
+#
+# Verify the test user can execute the zfs utilities.  This may not
+# be possible due to default permissions on the user home directory.
+# This can be resolved granting group read access.
+#
+# chmod 0750 $HOME
+#
+user_run $PUSER zfs list
+if [ $? -ne 0 ]; then
+	log_unsupported "Test user $PUSER cannot execute zfs utilities"
+fi
+
+DISK=${DISKS%% *}
+default_setup_noexit $DISK
+
+#
+# Need chattr/lsattr utils to set/show project ID. Verify whether the
+# e2fsprogs supports related commands options or not.
+#
+lsattr -pd $TESTDIR > /dev/null 2>&1
+if [ $? -ne 0 ]; then
+	log_unsupported "Current e2fsprogs does not support set/show project ID"
+fi
```

Comment:
> so all the project quota related test cases hould be skipped

I was suggesting that only the test case which depend on `lsattr -pd` and `chattr -P` be skipped.  Several of the test cases don't depend on these utilities since they are testing `zfs(8)` or `zpool(8)` functionality.  I think there's still value in running those tests to help prevent regressions on older platforms.

For example, CentOS 7 doesn't ship updated utilities for project quota but the following test cases still do pass as expected.  It'd be nice to run them.

```
Test: projectquota/projectquota_002_pos (run as root) [00:08] [PASS]
Test: projectquota/projectquota_004_neg (run as root) [00:01] [PASS]
Test: projectquota/projectquota_005_pos (run as root) [00:00] [PASS]
Test: projectquota/projectquota_007_pos (run as root) [00:01] [PASS]
Test: projectquota/projectquota_008_pos (run as root) [00:01] [PASS]
Test: projectquota/projectquota_009_pos (run as root) [00:01] [PASS]
```

> I do not thing it will affect other non-projectquota related test cases, right?

Right.